backend/app/services/api/filterer.rb

```diff
@@ -0,0 +1,44 @@
+module API
+  class Filterer
+    attr_accessor :query, :filters
+
+    ENUM_FILTERS = %i[category impact sdg instrument_type ticket_size]
+
+    def initialize(query, filters)
+      @query = query
+      @filters = filters
+    end
+
+    def call
+      filter_by_enums
+      filter_by_only_verified
+      query
+    end
+
+    private
+
+    def filter_by_enums
+      pluralize(filters.slice(*ENUM_FILTERS)).slice(*column_names).each do |filter_key, filter_value|
+        sql = ActiveRecord::Base.sanitize_sql ["ARRAY[#{filter_key}]::text[] && ARRAY[?]::text[]", Array.wrap(filter_value)]
+        self.query = query.where sql
+      end
+    end
+
+    def filter_by_only_verified
+      return unless filters[:only_verified] && column_names.include?("review_status")
+
+      self.query = query.where review_status: :approved
```

Comment:
I believe this is not what it needs to do. The only_verified filter applies to projects and open calls and it should work on the "trusted" column (we had a few iterations on the name... 😅 ).

But, as a matter of fact this brings up something else, i.e. in case of project developers and investors we have the `review_status` flag and when showing search results, we should only return those who are `approved`. I imagine this endpoint will be used for public search only, so perhaps we could add a `filter_by_review_status` with condition hardcoded in it, i.e. not controlled by parameters?

Reply:
Aha :), thanks for clarification, will fix it according to your recommendation.

Reply:
Yes, approved scope should be applied in controller before filters.